model/jpa/src/main/java/org/keycloak/models/jpa/UserAdapter.java

```diff
@@ -196,14 +196,22 @@ private void persistAttributeValue(String name, String value) {
 
     @Override
     public void removeAttribute(String name) {
-        List<UserAttributeEntity> toRemove = new ArrayList<>();
+        List<UserAttributeEntity> customAttributesToRemove = new ArrayList<>();
         for (UserAttributeEntity attr : user.getAttributes()) {
             if (attr.getName().equals(name)) {
-                toRemove.add(attr);
+                customAttributesToRemove.add(attr);
             }
         }
 
-        if (toRemove.isEmpty()) {
+        if (customAttributesToRemove.isEmpty()) {
```

Comment:
This branch makes sense from my POV, but I don't get how it relates to issue #27797.
Is this an additional improvement?

Reply:
Yeah, it is. We were missing removing the "root" attributes from the user but only custom attributes. Every other method related to managing attributes do something similar. So I just doing the same here for the sake of consistency.